### DISCUSSION TITLE
Best detector for use with Double-Take and Frigate on a Raspberry Pi?

### DISCUSSION THREAD

**Discussion by jerome71** [blakeblackshear/frigate#14109] [General Support]
Hi.
I have (successfully I think 😊) set up Frigate on a Raspberry Pi 5 8 GB + Coral USB TPU. 8 cameras, including 3 on a different location, integrated thanks to Tailscale. Some H264 re-encoding (CPU only on the Pi 5) that is responsible for most of the CPU usage.
It works great, and much better than I expected from such a setup.
The Pi still had 50% CPU power available.

I have now set up Double Take, with DeepStack. On the same Raspberry Pi. I get some results - they are not great (which I expected because the cameras are usually far from the subjects, so they appear in quite low resolution).

My question is: **what would be your recommendation be for a detector for face recognition, supported by Double Take, that works on a Raspberry Pi?**
My understanding is that Compreface is X64 only, so does not work on the Pi's ARM64.
So should I try CodeProject.AI Server? Or Facebox?
Or should I stick with Deepstack?

Bonus question: is there a way I can use the Coral USB TPU in both Frigate and the Double Take detector?
I tried using the Coral USB TPU simultaneously in 2 Frigate instances and could not make it work.

Thanks in advance! 😊🙏

**Comment by molnarti** [comment]
hey, what did you end up with? i am in the same situation, both all detectors (deepstack, facebox or compreface) seem to be dead with last releases 2 years ago...

also, how did you integrate your offsite cameras? where are you running the tailscale nodes on them?

**Comment by elektrinis** [comment]
just confirming that these detectors (deepstack, facebox or compreface) are dead and actually really awful, unusable. They detect my wife as me etc.
It looks like open source face detection is dead for now. On frigate/HA at least.

**Reply by NickM-27** [reply]
I disagree, using compreface it worked really well for me. Compreface worked with multiple different models and many users made the mistake of not customizing which model they were using.

And also, Frigate 0.16 already has face recognition and ALPR implemented.

**Reply by bozzio83** [reply]
> And also, Frigate 0.16 already has face recognition and ALPR implemented

So I won't need a third party face recognition app?  
Will there be a way to use the immich api for face recognition?

**Reply by NickM-27** [reply]
> So I won't need a third party face recognition app?

correct

> Will there be a way to use the immich api for face recognition?

Not sure what API you are referring to, https://immich.app/docs/api/get-faces/

**Reply by bozzio83** [reply]
> correct

That's great news!
 
> Not sure what API you are referring to, https://immich.app/docs/api/get-faces/

I thought there was a way to use their API with this.  I guess what I read was wrong.  But thank you for the information :)

**Comment by lngarrett** [comment]
@NickM-27 do you have any guidance on getting good results on a doorbell cam with compreface? I just set it up tonight. Also, I'm open to just hopping on a 0.16 branch if that's where things are going in a few months and I can help test. I saw a comment earlier that official 0.16 RCs might not be til later this year, so I was exploring DT/compreface in the meantime.

**Reply by NickM-27** [reply]
The docs are likely still applicable https://deploy-preview-16390--frigate-docs.netlify.app/configuration/face_recognition

**Reply by bozzio83** [reply]
Does retention for faces work the same as for like record?
```
face_recognition:
  enabled: true
  retain:
    days: 7
```

Just so the list of untrained faces doesn't keep growing?

**Reply by NickM-27** [reply]
No, currently they still stay indefinitely
